File: extract.js
import AdmZip from 'adm-zip';

try {
  const zip = new AdmZip('dist/mym-source.zip');
  const zipEntries = zip.getEntries();
  console.log(`Found ${zipEntries.length} entries in dist/mym-source.zip`);
  
  // Find AdminDashboard.tsx
  const entry = zipEntries.find(e => e.entryName.includes('AdminDashboard.tsx'));
  if (entry) {
    console.log(`Found AdminDashboard.tsx in zip: ${entry.entryName}`);
    zip.extractEntryTo(entry, 'temp_extracted', false, true);
    console.log('Extracted successfully to temp_extracted!');
  } else {
    console.log('AdminDashboard.tsx not found in zip');
  }
} catch (e) {
  console.error('Error reading zip:', e);
}

File: extract.py
import zipfile
import os

zips = [
    'public/mym-source.zip',
    'dist/mym-source.zip',
    'dist/mym-website.zip',
    'public/mym-website.zip'
]

for zip_path in zips:
    if not os.path.exists(zip_path):
        print(f"{zip_path} does not exist")
        continue
    print(f"Checking {zip_path}...")
    try:
        with zipfile.ZipFile(zip_path, 'r') as z:
            for name in z.namelist():
                if 'AdminDashboard.tsx' in name:
                    print(f"Found in {zip_path}: {name}")
                    z.extract(name, 'extracted_temp')
                    print(f"Extracted {name} to extracted_temp!")
    except Exception as e:
        print(f"Error checking {zip_path}: {e}")
